app/v3/service/internal/keeper/binding.go

```diff
@@ -223,6 +251,78 @@ func (k Keeper) GetServiceBinding(ctx sdk.Context, serviceName string, provider
 	return svcBinding, true
 }
 
+// ParsePricing parses the given string to Pricing
+func (k Keeper) ParsePricing(ctx sdk.Context, pricing string) (p types.Pricing, err sdk.Error) {
+	var rawPricing types.RawPricing
+	if err := json.Unmarshal([]byte(pricing), &rawPricing); err != nil {
+		return p, types.ErrInvalidPricing(k.codespace, fmt.Sprintf("failed to unmarshal the pricing: %s", err))
+	}
+
+	var coins sdk.Coins
+
+	coinStrs := strings.Split(rawPricing.Price, ",")
+	for _, coinStr := range coinStrs {
+		unitName, amtStr, _ := sdk.ParseCoinParts(coinStr)
+		if err != nil {
+			return p, types.ErrInvalidPricing(k.codespace, fmt.Sprintf("failed to parse the pricing: %s", err))
+		}
+
+		amt, err := sdk.NewDecFromStr(amtStr)
+		if err != nil {
+			return p, types.ErrInvalidPricing(k.codespace, fmt.Sprintf("failed to parse the pricing: %s", err))
+		}
+
+		if unitName == sdk.Iris {
+			coins = coins.Add(sdk.NewCoins(sdk.NewCoin(
+				sdk.IrisAtto,
+				amt.Mul(sdk.NewDecFromInt(sdk.NewIntWithDecimal(1, sdk.AttoScale))).TruncateInt(),
+			)))
+		} else {
+			token, err := k.ak.GetToken(ctx, unitName)
+			if err != nil {
+				return p, types.ErrInvalidPricing(k.codespace, fmt.Sprintf("failed to parse the pricing: %s", err))
```

Comment:
invalid coin denom?

Reply:
invalid price?  No coin or denom in the pricing schema

Reply:
the `ErrInvalidPricing` already contains `invalid pricing:`